sdks/java/extensions/kryo/src/main/java/org/apache/beam/sdk/extensions/kryo/KryoCoder.java

```diff
@@ -21,106 +21,211 @@
 import com.esotericsoftware.kryo.KryoException;
 import com.esotericsoftware.kryo.io.InputChunked;
 import com.esotericsoftware.kryo.io.OutputChunked;
-import com.google.common.annotations.VisibleForTesting;
+import java.io.EOFException;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import org.apache.beam.sdk.coders.AtomicCoder;
 import org.apache.beam.sdk.coders.CoderException;
-import org.apache.beam.sdk.coders.CustomCoder;
+import org.apache.beam.sdk.options.PipelineOptions;
 
 /**
  * Coder using Kryo as (de)serialization mechanism. See {@link KryoCoderProvider} to get more
- * details of how to use it
+ * details about usage.
+ *
+ * @param <T> type of element coder can handle
  */
-public class KryoCoder<T> extends CustomCoder<T> {
+public class KryoCoder<T> extends AtomicCoder<T> {
```

Comment:
Not sure that 'KryoCoder' is an `AtomicCoder` since docs states:
> All atomic coders of the same class are considered to be equal to each other. As a result, an {@link AtomicCoder} should have no associated configuration (instance variables, etc).

Reply:
fixed 👍